### PR TITLE
Refactor delegates redux logic - Closes #2023

### LINKED ISSUE #2023: Review and refactor the actions, reducers and middleware for delegates.
### Descriptions
This is a child ticket for #2020 .
With a short review over the code, we can find several redundant or obsolete actions. accordingly there are cases defined in the reducers switch-case statement which can be refactored or even removed. 
The naming of the actions seem to need revision.

More specifically please remove`delegateRegisteredFailure`. Handle this using React components.
With the new version of the multistep component we can easily handle this.

### LINKED ISSUE #2027: Review and refactor the actions, reducers and middleware for votes
### Description
This is a child ticket for #2020 .
With a short review over the code, we can find several redundant or obsolete actions. accordingly there are cases defined in the reducers switch-case statement which can be refactored or even removed.
The redux store must not handle the transitional states. Any action storing data that only handles such a situation must be removed.

### PR DESCRIPTION
**Pull request by slaweet**
### What issue have I solved?

#2023

### How have I implemented/fixed it?

There are TODOs in the code describing what should be changed in the initial commits:
https://github.com/LiskHQ/lisk-hub/pull/2099/commits/97d833677e18bd3debd4adac9ccae4360eacc62a
https://github.com/LiskHQ/lisk-hub/pull/2099/commits/ed4502001180c1b73a24acabb40f5791ed1e9461
https://github.com/LiskHQ/lisk-hub/pull/2099/commits/0ac0a9e065d20a58df4c49643670804bfcad87ad
https://github.com/LiskHQ/lisk-hub/pull/2099/commits/a2aa019d502acb934a85fc0f9cde83baa5de88df
Rest of the commits resolves those todos.

### How has this been tested?

Check that e2e tests still pass.

### Review checklist
- The PR follows our [Test guide](/LiskHQ/lisk-hub/blob/development/docs/TEST_GUIDE.md)
- The PR follows our [CSS guide](/LiskHQ/lisk-hub/blob/development/docs/CSS_GUIDE.md)

### CODE REVIEW

**Comment by slaweet** [comment]
Good point @osvaldovega. I added two more TODOs in https://github.com/LiskHQ/lisk-hub/pull/2099/commits/a2aa019d502acb934a85fc0f9cde83baa5de88df to address your comment.

File: src/utils/voting.js [hunk 1]
```diff
@@ -1,29 +1,55 @@
...
+});
+
+export const getVotingError = (votes, account) => {
+  let error;
+  if (account.balance < Fees.vote) {
+    error = i18next.t('Not enough LSK to pay for the transaction.');
+  } else if (getTotalVotesCount(votes) > votingConst.maxCountOfVotes) {
+    error = i18next.t('Max amount of delegates in one voting exceeded.');
+  }
+  return error;
```
Comment (osvaldovega) [inline src/utils/voting.js]: I think you dont need the error variable, maybe something like this.

```
if (account.balance < Fees.vote) return i18next.t('Not enough LSK to pay for the transaction.');
if (getTotalVotesCount(votes) > votingConst.maxCountOfVotes) return i18next.t('Max amount of delegates in one voting exceeded.');
```

but is just a suggestion if not like this is ok too

Comment (slaweet) [reply src/utils/voting.js]: Your suggestion results in eslint error: `consistent-return: Expected to return a value at the end of arrow function.`

File: src/actions/search.js [hunk 2]
```diff
@@ -71,13 +71,13 @@ export const fetchVotedDelegateInfo = (votes, {
     }
   };
 
-const searchVotes = ({ address, offset, limit }) =>
+const searchVotes = ({ address }) =>
   async (dispatch, getState) => {
     const liskAPIClient = getState().peers.liskAPIClient;
```
[the remark this answers is not in the document]
Comment (slaweet) [reply src/actions/search.js]: Yes, it should use the network, but there is a separate ticket for search actions: https://github.com/LiskHQ/lisk-hub/issues/2025 I was just adjusting it here to the changes in getVotes API util.

**Comment by slaweet** [comment]
Thank you @osvaldovega for your feedback. Good points. 

I either fixed or put an explanation to each of them. Please review again

File: src/actions/search.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (osvaldovega) [reply src/actions/search.js]: oh ok got it, thx Vit

**Pull request merged**